pkg/rbd/rbd_util.go

```diff
@@ -848,23 +856,16 @@ func resizeRBDImage(rbdVol *rbdVolume, newSize int64, cr *util.Credentials) erro
 	return nil
 }
 
-func getVolumeName(volID string) (string, error) {
+func ensureEncryptionMetadataSet(ctx context.Context, cr *util.Credentials, rbdVol *rbdVolume) error {
 	var vi util.CSIIdentifier
 
-	err := vi.DecomposeCSIID(volID)
+	err := vi.DecomposeCSIID(rbdVol.VolID)
 	if err != nil {
-		err = fmt.Errorf("error decoding volume ID (%s) (%s)", err, volID)
-		return "", ErrInvalidVolID{err}
+		err = fmt.Errorf("error decoding volume ID (%s) (%s)", err, rbdVol.VolID)
```

Comment:
the error should be at last, as you touched this line can you please fix it here?